packages/react-component-library/src/fragments/NotificationPanel/NotificationPanel.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useRef, useState } from 'react'
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group'
+
+import { Popover } from '../../components'
+import { Bell } from '../../icons'
+
+export interface NotificationPanelProps {
+  buttonClassName?: string
+  className?: string
+  notificationPlacement?: 'right' | 'below'
+  onShow?: () => void
+  onHide?: () => void
+  unreadNotification?: boolean
+}
+
+const POPOVER_WIDTH = 335
+
+export function getNotificationPositionOnRight(
+  element: Element | any,
+  elementWindow: any = window
+): PositionType {
+  const elemRect = element.getBoundingClientRect()
+  const bottom =
+    elementWindow.innerHeight - elemRect.bottom - 8 + window.pageYOffset
+  const left = elemRect.left + elemRect.width + 18 + window.pageXOffset
+
+  return { bottom, left }
+}
+
+export function getNotificationPositionBelow(
+  element: Element | any
+): PositionType {
+  const elemRect = element.getBoundingClientRect()
+
+  const left =
+    elemRect.left - POPOVER_WIDTH + elemRect.width + window.pageXOffset + 5
+  const top = elemRect.bottom + 3 + window.pageYOffset
+
+  return { left, top }
+}
+
+export const NotificationPanel: React.FC<NotificationPanelProps> = ({
+  buttonClassName = '',
+  children,
+  className = '',
+  notificationPlacement = 'right',
+  onShow,
+  onHide,
+  unreadNotification = false,
+}) => {
+  const notificationArrowPosition =
+    notificationPlacement === 'right' ? 'left_bottom' : 'top_right'
+
+  const [showNotifications, setShowNotifications] = useState(false)
+  const [notificationPosition, setNotificationPosition] = useState()
+  const node = useRef()
+
+  function documentClick(event: Event) {
+    // workaround for undefined error in typescript
```

Comment:
Anyone know what this is about? Is this to deal with browser APIs missing in node when doing SSR?